synapse/rest/client/v2_alpha/account.py

```diff
@@ -485,12 +542,126 @@ def on_POST(self, request):
         if existing_user_id is not None:
             raise SynapseError(400, "MSISDN is already in use", Codes.THREEPID_IN_USE)
 
+        if not self.hs.config.account_threepid_delegate_msisdn:
```

Comment:
again this doesn't seem to match the PR description. Can we pull it out?

Reply:
It's part of MSC2290, I can move it to that PR.

Reply:
https://github.com/matrix-org/synapse/commit/c8a3a1e4357d28a365f434c8ec3f835afb9c06c1